### ISSUE TITLE
Protect release branches

### ISSUE THREAD

**Issue by ggwpez** [paritytech/pr-custom-review#134]
All substrate branches with prefix `polkadot-v` and all Polkadot branches with prefix `release-v` should require approval from @coderobe or @chevdor. 

I could not find such an option in the GitHub branch protection, but we need this since the release branches keep getting pushed without coordination.
```[tasklist]
### Tasks
```

**Comment by mordamax** [comment]
@ggwpez in gh team/enterprise plan they seem to have rulesets, which can make this happen. PR custom review, is only for PRs, and it can't prevent push or creation

this is test from https://github.com/paritytech-stg/polkadot-sdk/settings/rules/55406
<img width="857" alt="image" src="https://github.com/paritytech/pr-custom-review/assets/1177472/f2359a6b-0d37-4d14-81d2-4c6d13835edb">

@paritytech-ci @lovelaced could you please take a look and help to configure this ?

**Comment by coderobe** [comment]
This would be extremely useful, both accidental and intentional circumvention happens way too frequently.
Whichever solution we have should allow @paritytech/release-engineering to push/approve

**Comment by lovelaced** [comment]
Right, if everyone agrees I can put this in place on both polkadot and substrate. There's a few more items I need to know the details of here:

![Screenshot_20230725-091818.png](https://github.com/paritytech/pr-custom-review/assets/8581604/37daec8a-1275-4d83-baa7-cf568fe35a1f)

**Comment by lovelaced** [comment]
Okay, rules have been created and are enforcing. Please let me know if something isn't behaving as expected

**Comment by lovelaced** [comment]
Currently each is resolving to these branches being enforced under the rulesets:

![Screenshot_20230725-093739.png](https://github.com/paritytech/pr-custom-review/assets/8581604/5d276530-fb9c-4b37-85d9-e3455d0613c2)

![Screenshot_20230725-093724.png](https://github.com/paritytech/pr-custom-review/assets/8581604/26d23197-89ef-43a3-9196-d067cd576856)

**Comment by coderobe** [comment]
i think you might have the targets the wrong way around

**Comment by lovelaced** [comment]
Oops so it is. Fixed

**Comment by coderobe** [comment]
Thanks 🙏

**Comment by ggwpez** [comment]
Looks like we cannot merge this now because of some rule https://github.com/paritytech/polkadot/pull/7571. 

<img width="844" alt="Screenshot 2023-08-02 at 11 57 58" src="https://github.com/paritytech/pr-custom-review/assets/10380170/b1f55f97-c085-4981-8e78-7a840a398a29">

I think this "Only allow users with bypass permission to update matching refs" means that only Admins can actually merge anything, or?

**Comment by coderobe** [comment]
> Looks like we cannot merge this now because of some rule [paritytech/polkadot#7571](https://github.com/paritytech/polkadot/pull/7571).
> 
> <img alt="Screenshot 2023-08-02 at 11 57 58" width="844" src="https://user-images.githubusercontent.com/10380170/257776653-b1f55f97-c085-4981-8e78-7a840a398a29.png">
> I think this "Only allow users with bypass permission to update matching refs" means that only Admins can actually merge anything, or?

No, releng can :)
merged ^^

**Comment by ggwpez** [comment]
Okay good, thanks!
